test/functional/test_site/expected/index.html

```diff
@@ -251,6 +251,7 @@ <h1 id="panel-without-heading-with-keyword">Panel without heading with keyword<a
             <h1 id="keyword-should-be-tagged-to-this-heading-not-the-panel-heading">Keyword should be tagged to this heading, not the panel heading<a class="fa fa-anchor" href="#keyword-should-be-tagged-to-this-heading-not-the-panel-heading" onclick="event.stopPropagation()"></a></h1>
             <p><span class="keyword">panel keyword</span></p>
           </panel>
+          <p></p>
```

Comment:
This occurs due to html not needing a closing `<p>` tag.
Previously, this is the _intermediate_ output after markdown-it renders the html.
It renders invalid html because it is subtly invalid markdown - `panel` is not a whitelisted html block tag and cannot terminate the paragraph rule (also resulting in #958):

```
<p><span class="keyword">panel keyword</span>
</panel>
</p>
```

when rerendering the html after having passed it through cheerio, the bottom lone `</p>` is invalid in `xmlMode`, and gets erased. when rendering without `xmlMode` however, it gets expanded into an empty `p` tag by htmlparser2's forgiving nature (there is no spec for this afaik). (line 226 of htmlparser2's Parser.js)

Ultimately, the extra `<p></p>` is due to miswritten markdown;
```
<span class="keyword">
panel keyword</span>
```

would be the correct way of not inserting any additional `<p>`'s at all (note the newline). (I've left this in for now as to explain the many similar cases in the 2103 site)